File: languages/python/collatz.py
# Collatz Conjecture in Python

def collatz_conjecture(n):
    steps = 0
    if n != 1:
        print(f"\nUser Input: {user_input}.\n\n{n} is not equal to 1.\nBegin calculation...")

    while n != 1:
        if n % 2 == 0:
            next_number = n // 2
            if next_number == 1:
                print(f"\nStep {(steps + 1)}\n--------\n{n} is even.\n({n} // 2) = {next_number}.\n{next_number} is equal to 1.\nCalculation complete.")
            else:
                print(f"\nStep {(steps + 1)}\n--------\n{n} is even.\n({n} // 2) = {next_number}.\n{next_number} is not equal to 1.\nContinue calculation...")
            n = next_number
        else:
            next_number = (n * 3) + 1
            if next_number == 1:
                print(f"\nStep {(steps + 1)}\n--------\n{n} is odd.\n(({n} * 3) + 1) = {next_number}.\n{next_number} is equal to 1.\nCalculation complete.")
            else:
                print(f"\nStep {(steps + 1)}\n--------\n{n} is odd.\n(({n} * 3) + 1) = {next_number}.\n{next_number} is not equal to 1.\nContinue calculation...")
            n = next_number
        steps += 1

    print(f"\n{user_input} reached 1 in {steps} steps.")

user_input = int(input("Enter a Positive Whole Number: "))

if user_input <= 0:
    print("Please enter a positive whole number.")
else:
    collatz_conjecture(user_input)

File: languages/cpp/collatz.cpp
#include <iostream>

void collatzConjecture(int n, int userInput) {
    int steps = 0;
    if (n != 1) {
        std::cout << "\nUser Input: " << userInput << ".\n\n" << n << " is not equal to 1.\nBegin calculation...\n";
    }
    while (n != 1) {
        if (n % 2 == 0) {
            int nextNumber = n / 2;
            if (nextNumber == 1) {
                std::cout << "\nStep " << (steps + 1) << "\n--------\n"
                          << n << " is even.\n(" << n << " / 2) = " << nextNumber
                          << ".\n" << nextNumber << " is equal to 1.\nCalculation complete.\n";
            } else {
                std::cout << "\nStep " << (steps + 1) << "\n--------\n"
                          << n << " is even.\n(" << n << " / 2) = " << nextNumber
                          << ".\n" << nextNumber << " is not equal to 1.\nContinue calculation...\n";
            }
            n = nextNumber;
        } else {
            int nextNumber = (n * 3) + 1;
            if (nextNumber == 1) {
                std::cout << "\nStep " << (steps + 1) << "\n--------\n"
                          << n << " is odd.\n((" << n << " * 3) + 1) = " << nextNumber
                          << ".\n" << nextNumber << " is equal to 1.\nCalculation complete.\n";
            } else {
                std::cout << "\nStep " << (steps + 1) << "\n--------\n"
                          << n << " is odd.\n((" << n << " * 3) + 1) = " << nextNumber
                          << ".\n" << nextNumber << " is not equal to 1.\nContinue calculation...\n";
            }
            n = nextNumber;
        }
        steps++;
    }

    std::cout << "\n" << userInput << " reached 1 in " << steps << " steps.\n";
}

int main() {
    int userInput;

    std::cout << "Enter a Positive Whole Number: ";
    std::cin >> userInput;

    if (std::cin.fail() || userInput <= 0) {
        std::cout << "Please enter a Positive Whole Number.\n";
    } else {
        collatzConjecture(userInput, userInput);
    }

    return 0;
}